Classes/Domain/Repository/AbstractRepository.php

```diff
@@ -43,20 +46,77 @@
  *             http://opensource.org/licenses/gpl-license.php
  *
  */
-abstract class AbstractRepository extends Repository {
+abstract class AbstractRepository extends Repository
+{
 
-	/**
-	 * @return QueryInterface
-	 */
-	protected function createQueryWithFallbackStoragePage() {
-		$query = $this->createQuery();
+    /**
+     * @var \TYPO3\CMS\Extbase\Service\TypoScriptService
+     * @inject
+     *
+     */
+    protected $typoScriptService;
 
-		$storagePageIds = $query->getQuerySettings()->getStoragePageIds();
-		$storagePageIds[] = 0;
+    /**
+     * @var array
+     */
+    protected $settings = [];
 
-		$query->getQuerySettings()->setStoragePageIds($storagePageIds);
+    /**
+     * @var array
+     */
+    protected $persistenceSettings = [];
 
-		return $query;
-	}
+    /**
+     *
+     */
+    public function initializeObject()
+    {
+        $ts = $this->getTypoScriptService()->convertTypoScriptArrayToPlainArray($GLOBALS['TSFE']->tmpl->setup);
+        $this->settings = $ts['plugin']['tx_typo3forum']['settings'];
+        $this->persistenceSettings = $ts['plugin']['tx_typo3forum']['persistence'];
+
+        if (isset($this->persistenceSettings['storagePid'])) {
+            $this->setDefaultQuerySettings(
+                $this->getQuerySettings()->setStoragePageIds(explode(',', $this->persistenceSettings['storagePid']))
+            );
+        }
+    }
+
+
+    /**
+     * @return QueryInterface
+     */
+    protected function createQueryWithFallbackStoragePage()
+    {
+        $query = $this->createQuery();
+
+        $storagePageIds = $query->getQuerySettings()->getStoragePageIds();
+        $storagePageIds[] = 0;
+
+        $query->getQuerySettings()->setStoragePageIds($storagePageIds);
+
+        return $query;
+    }
+
+
+    /**
+     * @return TypoScriptService
+     */
+    protected function getTypoScriptService()
+    {
+        if (is_null($this->typoScriptService)) {
+            $this->typoScriptService = $this->objectManager->get('TYPO3\\CMS\\Extbase\\Service\\TypoScriptService');
+        }
```

Comment:
Directly instantiating the `TypoScriptService` seems unnecessary, when you're also getting it via dependency injection.